cylc/flow/command_validation.py

```diff
@@ -0,0 +1,227 @@
+from cylc.flow.flow_mgr import FLOW_ALL, FLOW_NEW, FLOW_NONE
+
+
+ERR_OPT_FLOW_VAL = "Flow values must be an integer, or 'all', 'new', or 'none'"
+ERR_OPT_FLOW_INT = "Multiple flow options must all be integer valued"
+ERR_OPT_FLOW_WAIT = (
+    f"--wait is not compatible with --flow={FLOW_NEW} or --flow={FLOW_NONE}"
+)
+
+
+def flow_opts(flows: List[str], flow_wait: bool) -> None:
+    """Check validity of flow-related CLI options.
+
+    Note the schema defaults flows to ["all"].
+
+    Examples:
+        Good:
+        >>> flow_opts(["new"], False)
+        >>> flow_opts(["1", "2"], False)
+        >>> flow_opts(["1", "2"], True)
+
+        Bad:
+        >>> flow_opts(["none", "1"], False)
+        Traceback (most recent call last):
+        cylc.flow.exceptions.InputError: ... must all be integer valued
+
+        >>> flow_opts(["cheese", "2"], True)
+        Traceback (most recent call last):
+        cylc.flow.exceptions.InputError: ... or 'all', 'new', or 'none'
+
+        >>> flow_opts(["new"], True)
+        Traceback (most recent call last):
+        cylc.flow.exceptions.InputError: ...
+
+    """
+    for val in flows:
+        val = val.strip()
+        if val in [FLOW_NONE, FLOW_NEW, FLOW_ALL]:
+            if len(flows) != 1:
+                raise InputError(ERR_OPT_FLOW_INT)
+        else:
+            try:
+                int(val)
+            except ValueError:
+                raise InputError(ERR_OPT_FLOW_VAL.format(val))
+
+    if flow_wait and flows[0] in [FLOW_NEW, FLOW_NONE]:
+        raise InputError(ERR_OPT_FLOW_WAIT)
+
+
+def prereqs(prereqs: Optional[List[str]]):
+    """Validate a list of prerequisites, add implicit ":succeeded".
+
+    Comma-separated lists should be split already, client-side.
+
+    Examples:
+        # Set multiple at once:
+        >>> prereqs(['1/foo:bar', '2/foo:baz'])
+        ['1/foo:bar', '2/foo:baz']
+
+        # --pre=all
+        >>> prereqs(["all"])
+        ['all']
+
+        # implicit ":succeeded"
+        >>> prereqs(["1/foo"])
+        ['1/foo:succeeded']
+
+        # Error: invalid format:
+        >>> prereqs(["fish"])
+        Traceback (most recent call last):
+        cylc.flow.exceptions.InputError: ...
+
+        # Error: invalid format:
+        >>> prereqs(["1/foo::bar"])
+        Traceback (most recent call last):
+        cylc.flow.exceptions.InputError: ...
+
+        # Error: "all" must be used alone:
+        >>> prereqs(["all", "2/foo:baz"])
+        Traceback (most recent call last):
+        cylc.flow.exceptions.InputError: ...
+
+    """
+    if prereqs is None:
+        return []
+
+    prereqs2 = []
+    bad: List[str] = []
+    for pre in prereqs:
+        p = prereq(pre)
+        if p is not None:
+            prereqs2.append(p)
+        else:
+            bad.append(pre)
+    if bad:
+        raise InputError(
+            "Use prerequisite format <cycle-point>/<task>:output\n"
+            "\n  ".join(bad)
+        )
```

Comment:
There's a subtle error on this line which was causing the problem.

It should be:

```
+ '\n  '.join(bad)
```

Without the `+` the error contained only `bad`.